sources/alpaquita-http.go

```diff
@@ -0,0 +1,88 @@
+package sources
+
+import (
+	"crypto/sha512"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/lxc/distrobuilder/shared"
+)
+
+type alpaquita struct {
+	common
+}
+
+func (s *alpaquita) Run() error {
+	baseURL, fname, err := s.getMiniroot(s.definition)
+	if err != nil {
+		return err
+	}
+
+	var fpath string
+	tarballURL := baseURL + fname
+	if s.definition.Source.SkipVerification {
+		fpath, err = s.DownloadHash(s.definition.Image,
+			tarballURL, "", nil)
+	} else {
+		fpath, err = s.DownloadHash(s.definition.Image,
+			tarballURL, tarballURL+".sha512", sha512.New())
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
Please move this into both the `if` and `else` chunk above. Otherwise we're at a high risk to add more logic to either of those chunks and mess up the error handling.